monai/metrics/occlusion_sensitivity.py

```diff
@@ -0,0 +1,120 @@
+    pad_val: float = 0.0,
+    margin: Union[int, Sequence] = 2,
+    n_batch: int = 128,
+) -> np.ndarray:
+    """
+    This function computes the occlusion sensitivity for a model's prediction
+    of a given image. The result is given as ``baseline`` (the probability of
+    a certain output) minus the probability of the output with the occluded
+    area. Therefore, in the output image, higher values mean there was a
+    greater the drop in certainty, indicating the occluded region was more
+    important in the decision process.
+
+    By occlusion sensitivity, we mean how the probability of a given
+    prediction changes as the occluded section of an image changes. This can
+    be useful to understand why a network is making certain decisions.
+
+    See:
+        R. R. Selvaraju et al. Grad-CAM: Visual Explanations from Deep Networks via Gradient-based Localization.
+    https://doi.org/10.1109/ICCV.2017.74
+
+    Args:
+        model: model to use for inference
+        image: image to test
+        label: classification label to check for changes (normally the true
+            label, but doesn't have to be)
+        pad_val: when occluding part of the image, which values should we put
+            in the image?
+        margin: we'll create a cuboid/cube around the voxel to be occluded. if
+            ``margin==2``, then we'll create a cube that is +/- 2 voxels in
+            all directions (i.e., a cube of 5 x 5 x 5 voxels). A ``Sequence``
+            can be supplied to have a margin of different sizes (i.e., create
+            a cuboid).
+        n_batch: number of images in a batch before inference.
+    """
+
+    # If necessary turn the label into a 1-element tensor
+    if isinstance(label, int):
+        label = torch.tensor([[label]], dtype=torch.int64).to(image.device)
+    # If the label is a tensor, make sure  there's only 1 element
+    elif label.numel() > 1:
+        raise RuntimeError("Expected 1 label.")
+
+    # Make sure the image is 4D tensor with 1 channel
+    while len(image.shape) < 4:
+        image = image.unsqueeze(0)
+    if image.shape[0] > 1:
+        raise RuntimeError("Expected 1 channel.")
+
+    # Get image shape
+    im_shape = image.shape[1:]
+
+    # Get baseline probability
+    baseline = model(image).detach()[0, label].item()
+
+    # Create some lists
+    batch_images_lst = []
+    batch_ids_lst = []
+
+    heatmap = torch.empty(0, dtype=torch.float32, device=image.device)
+    # Loop 1D over image
+    for i in trange(image.numel()):
+        # Get corresponding ND index
+        idx = np.unravel_index(i, im_shape)
+        # Get min and max index of bounding box.
+        min_idx = [max(0, i - margin) for i in idx]
+        max_idx = [min(j, i + margin) for i, j in zip(idx, im_shape)]
+
+        # Clone and replace target area with `pad_val`
+        occlu_im = image.clone()
+        occlu_im[..., min_idx[0] : max_idx[0], min_idx[1] : max_idx[1], min_idx[2] : max_idx[2]] = pad_val
```

Comment:
For arbitrary `ndim`

```suggestion
        occlu_im[(...,) + tuple(slice(i, j) for i, j in zip(min_idx, max_idx))] = pad_val
```